site/docs/develop/spec.md

```diff
@@ -48,8 +48,15 @@ Therefore, the Iceberg table spec should be considered subject to change in the
 
 ## Version Store
 
+There are several expections of a version store. Some are listed below
+
 !!! warning
-    todo
+    still missing some conditions
+    
+### Conflict Resolution
+The API passes an ‘expectedHash’ parameter when it modifies a key. This is the commit that the client thinks is the most up to date (its HEAD). A backend will check to see if the key has been modified since that ‘expectedHash' and if it has will reject the requested modification with FAILED_PRECONDITION. This is basically an optimistic lock that accounts for the fact that the commit hash is global and nessie branch could have moved on from ‘expectedHash’ without modifying the key in question.
```

Comment:
```suggestion
The API passes an ‘expectedHash’ parameter when it modifies a key. This is the commit that the client thinks is the most up to date (its HEAD). A backend will check to see if the key has been modified since that ‘expectedHash' and if, it will reject the requested modification with FAILED_PRECONDITION. This is basically an optimistic lock that accounts for the fact that the commit hash is global and nessie branch could have moved on from ‘expectedHash’ without modifying the key in question.
```